### PR TITLE
Use modern cmake to fix compiling on ROS2

### PR DESCRIPTION
**Pull request by JafarAbdi**
Fix: https://github.com/ros-planning/moveit2/issues/588#issuecomment-892776014

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,10 +42,10 @@ ament_target_dependencies(${PROJECT_NAME} PUBLIC
 )
 
 install(TARGETS ${PROJECT_NAME}
-  EXPORT ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}Targets
```
Comment (vatanaksoytezer) [inline CMakeLists.txt:45]:
```suggestion
  EXPORT export_${PROJECT_NAME}
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -90,8 +85,7 @@ if(BUILD_TESTING)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-ament_export_include_directories(include)
-ament_export_libraries(${PROJECT_NAME})
+ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
```
Comment (vatanaksoytezer) [inline CMakeLists.txt:88]:
```suggestion
ament_export_targets(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
```

File: CMakeLists.txt (hunk 1 above)
Comment (vatanaksoytezer) [reply CMakeLists.txt:45]: I think this usage is more common

File: CMakeLists.txt (hunk 2 above)
Comment (vatanaksoytezer) [reply CMakeLists.txt:88]: Also this

**Review by vatanaksoytezer — APPROVED** [review]
I think this is much needed. I like the modern style and it is much more compact. See one small nit below

**Review by tylerjw — APPROVED** [review]
Approve assuming you take the changes vatan suggested.

File: CMakeLists.txt (hunk 1 above)
Comment (JafarAbdi) [reply CMakeLists.txt:45]: I don't know where the name `export_${PROJECT_NAME}` came from, but multiple known libraries ([gtest](https://github.com/google/googletest/blob/master/googletest/CMakeLists.txt#L102), [fmt](https://github.com/fmtlib/fmt/blob/master/CMakeLists.txt#L315), and [nlohmann_json](https://github.com/nlohmann/json/blob/develop/CMakeLists.txt#L53)) use either `{PROJECT_NAME}Targets` or `${PROJECT_NAME}-targets` which IMO is a better name for an export set

Comment (vatanaksoytezer) [reply CMakeLists.txt:45]: Idk either but that's what people have been using all over moveit2 (see https://github.com/ros-planning/moveit2/search?q=export_%24)

Comment (tylerjw) [reply CMakeLists.txt:45]: the export_ thing came from rclcpp, I copied it into moveit2.

Comment (tylerjw) [reply CMakeLists.txt:45]: I take that back, rclcpp doesn't do that... I think it came from the ros2 documentation on ament cmake... but now I can't find it.

Comment (tylerjw) [reply CMakeLists.txt:45]: https://docs.ros.org/en/foxy/Guides/Ament-CMake-Documentation.html#building-a-library

That Is where the pattern came from.  If `{PROJECT_NAME}Targets` or `${PROJECT_NAME}-targets` is more standard, please change it in moveit2 and elsewhere.

**Comment by tylerjw** [comment]
@JafarAbdi https://github.com/ros2/ros2_documentation/pull/1797

**Pull request merged**